--- a.c	2022-04-17 15:02:57.415102988 -0300
+++ b.c	2022-04-18 12:24:22.309849517 -0300
@@ -12,7 +12,7 @@
 void handler(int sig, siginfo_t *info, void *ucontext)
 {
     ucontext_t *context = (ucontext_t *) ucontext;
-    struct user_regs_struct *regs = (void *) context->uc_mcontext.gregs;
+    struct user_regs_struct *regs = (void *) context->uc_mcontext.gregs[REG_RIP];
     regs->rip = (long) test;
 }
 
